content/en/blog/2023/otel-end-user-q-and-a-series-otel-and-graphql.md

```diff
@@ -0,0 +1,346 @@
+
+### JavaScript Maintenance
+
+OpenTelemetry has a small number of language maintainers, and as a result, they
+don’t necessarily have enough cycles to work on all the things. Thus, they
+currently focus on keeping up with spec changes to update the SDK and API. This
+means that they often don’t have time (and sometimes not even the expertise) to
+manage the contrib repos (e.g. GraphQL). This is a known problem, and there is
+currently no solution in place. The OpenTelemetry Community welcomes any
+suggestions for improvement!
+
+There is also a huge focus on
+[stabilizing semantic conventions](https://docs.google.com/document/d/1ghvajKaipiNZso3fDtyNxU7x1zx0_Eyd02OGpMGEpLE/edit#),
+and as part of that effort, maintainers plan to go through the existing
+instrumentation packages and to make sure that they’re all up to date with the
+latest conventions. While it’s very well-maintained for certain languages, such
+as Java, that is not the case for other languages, such as Node.JS.
+
+JavaScript environments are akin to the Wild West of Development due to:
+
+- Multiple facets: web side vs server side
+- Multiple languages: JavaScript, TypeScript, Elm
+- Two similar, but different server-side runtimes: Node.JS and
+  [Deno](https://deno.land)
+
+One of J’s suggestions is to treat OTel Javascript as a hierarchy, which starts
+with a Core JavaScript team that splits into two subgroups: front-end web group,
+and back-end group. Front-end and back-end would in turn split. For example, for
+the back-end, have a separate Deno and Node.JS group.
+
+Another suggestion is to have a contrib maintainers group, separate from core
+SDK and API maintainers group.
+
+### JavaScript Contributions
+
+Making OpenTelemetry JavaScript contributions has been slow-moving at times,
+specifically around plug-ins. Much of the plug-in maintenance relies on the
+plug-in’s original owner; however, the original owner is gone in many cases, or
+maintainers don’t check GitHub very frequently, and as a result, movement on
+some pull requests (PRs) is very slow. One way to mitigate this is to get
+contributors more involved, which could potentially help bring in more
+contributors.
+
+### Documentation
+
+J and his team have also experienced some challenges with documentation, noting
+that there are some gaps in the online docs:
+
+- Under metrics for JavaScript there is no mention of the Observable Guage at
+  all. J had to go into the code to find it.
+- There are some short, very high-level metric API examples. Those examples
+  currently don't show which libraries you need to bring in. It also doesn't
+  talk about how to export items.
+- In .NET it is very hard to keep a trace going in your work due to all the
+  async/await and it jumping between threads. .NET docs lack some detail around
+  context propagation in this particular scenario.
+
+.NET docs and keeping context. I found a bunch of instrumentation but I noticed
+it has a hard time keeping the context around.
```

Comment:
This paragraph seems odd. The first sentence is incomplete.

Reply:
> This paragraph seems odd. The first sentence is incomplete.

My bad - that should've been removed. Deleted.

Reply:
please "resolve" the conversation